Source/WebCore/page/UserContentProvider.cpp

```diff
@@ -93,16 +93,49 @@ void UserContentProvider::invalidateInjectedStyleSheetCacheInAllFramesInAllPages
 }
 
 #if ENABLE(CONTENT_EXTENSIONS)
-static bool contentRuleListsEnabled(const DocumentLoader& documentLoader)
+static DocumentLoader* mainDocumentLoader(DocumentLoader& loader)
 {
-    if (auto frame = documentLoader.frame()) {
+    if (auto frame = loader.frame()) {
         if (frame->isMainFrame())
-            return documentLoader.userContentExtensionsEnabled();
-        if (auto mainDocumentLoader = frame->mainFrame().loader().documentLoader())
-            return mainDocumentLoader->userContentExtensionsEnabled();
+            return &loader;
+        return frame->mainFrame().loader().documentLoader();
+    }
+    return nullptr;
+}
+
+static ContentExtensions::ContentExtensionsBackend::RuleListFilter ruleListFilter(DocumentLoader& documentLoader)
+{
+    RefPtr mainLoader = mainDocumentLoader(documentLoader);
+    if (!mainLoader) {
+        return [](const String&) {
+            return ContentExtensions::ContentExtensionsBackend::ShouldSkipRuleList::No;
+        };
     }
 
-    return true;
+    return std::visit(WTF::makeVisitor([](const HashSet<String>& disabledRuleLists) -> ContentExtensions::ContentExtensionsBackend::RuleListFilter {
```

Comment:
I think this trailing return type is unnecessary.

Reply:
Ah, I had tried to remove these, but got:

```
error: static_assert failed due to requirement '__all<false>::value' "`std::visit` requires the visitor to have a single return type."
```

In lieu of the trailing return types, I wrapped these in a `ContentExtensions::ContentExtensionsBackend::RuleListFilter { }`, which takes care of the above build error.